File: apps/api/src/tariff.ts
import type { ResourceType, Tariff } from "./types.js";

export function computeCostEur(params: { resourceType: ResourceType; delta: number; tariff: Tariff }) {
  const { delta, tariff } = params;
  const base = delta * tariff.pricePerUnit;
  const total = base * (1 + tariff.vatRate);
  return total;
}

File: server/tariff.py
from __future__ import annotations

from .store import ResourceType, Tariff


def compute_cost_eur(resource_type: ResourceType, delta: float, tariff: Tariff) -> float:
    base = delta * tariff.pricePerUnit
    total = base * (1.0 + tariff.vatRate)
    return float(total)

